### PR TITLE
[TFA] Fixes to stabilise the ok to upgrade tests

### PR DESCRIPTION
**Pull request by s-vipin**
TFA fixes 

stabilization of holistic upgrade tests (tests/rados/test_osd_ok_to_upgrade_holistic_tests.py)
This file has the bulk of the fixes:

MGR failover resilience — Removed the assumption that the active MGR stays the same throughout the test. Instead of querying a single active MGR node for logs, the code now iterates over all MGR nodes and greps logs from all ceph-mgr.*.log files on each host.
Log rotation fix — rotate_logs() now targets all MGR-role nodes instead of just the active one.
Added --automatically-accept-license flag to the upgrade command.
Retry logic for ceph osd metadata — The downgrade helper now retries up to 5 times (with 20s sleep) when ceph osd metadata returns a non-list type or throws an exception, instead of failing immediately.
Typo fix — Changed remove_config to remove_configd when removing container_image during OSD downgrade.
Improved assertion messages — Error messages now reference "MGR host(s)" for clearer failure diagnostics.

Include missed negative test scenario (tier-2_rados_test_osd_ok_to_upgrade.yaml)
Added a new test entry ok-to-upgrade-negative-scenarios (CEPH-83632279) covering missing params, invalid bucket type/name, version format validation, convergence factor boundary values, and empty bucket.

Disabled known-buggy test cases (suite YAML files)
tier-2_rados_test-bugfixes.yaml — Commented out the Mon connection scores test due to bug [IBMCEPH-16972](https://ibm-ceph.atlassian.net/browse/IBMCEPH-16972).
tier-2_rados_test_osd_deployment_workflows.yaml — Commented out OSD deployment scenarios 5 & 6 (both spec and CLI variants) due to bug [IBMCEPH-13839](https://ibm-ceph.atlassian.net/browse/IBMCEPH-13839). These are the raw data+db and raw data+db+wal scenarios.

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by s-vipin** [comment]
Pass logs :- http://magna002.ceph.redhat.com/cephci-jenkins/vipin-runs/vips_automation_20260723-110143/ 
please note some tests are failing due to "osd metadata error". I will include retry for it. Other failing tests are now passing.

File: suites/tentacle/rados/tier-2_rados_test-bugfixes.yaml [hunk 1]
```diff
@@ -182,12 +182,13 @@ tests:
       polarion-id: CEPH-83594645
       desc: Ensure client connection over v1 port does not crash
 
-  - test:
-      name: Mon connection scores testing
-      desc: Verification of mon connection scores in different scenarios
-      module: test_mon_connection_scores.py
-      polarion-id: CEPH-83602911
-      comments: Intermittent bug 2357894
+  # Bug exists in 9.1z1 :- https://ibm-ceph.atlassian.net/browse/IBMCEPH-16972
```
Comment (s-vipin) [inline suites/tentacle/rados/tier-2_rados_test-bugfixes.yaml]: We might want to check if 9.2 also has same behaviour

**Review by neha-gangadhar — DISMISSED** [review]
lgtm

**Comment by neha-gangadhar** [comment]
@s-vipin please fix minor tox issue.

File: suites/tentacle/rados/tier-2_rados_test-bugfixes.yaml [hunk 2]
```diff
@@ -182,12 +182,13 @@ tests:
...
+  # - test:
```
Comment (s-vipin) [inline suites/tentacle/rados/tier-2_rados_test-bugfixes.yaml]: Harsh - move test to bottom

File: suites/tentacle/rados/tier-2_rados_test_osd_deployment_workflows.yaml [hunk 3]
```diff
@@ -169,25 +169,26 @@ tests:
...
+  # Bug - https://ibm-ceph.atlassian.net/browse/IBMCEPH-13839
+  # - test:
+  #     name: 5 - OSD deployment tests - raw data and db device - spec file
+  #     module: osd_deploy_tests.py
+  #     polarion-id: CEPH-83632255
+  #     config:
+  #       deployment_type: spec
+  #       test_scenarios:
+  #         - scenario5
+  #     desc: OSD deployment tests - raw data and db device - spec file
```
Comment (s-vipin) [inline suites/tentacle/rados/tier-2_rados_test_osd_deployment_workflows.yaml:180]: Include comments for commented tests

Comment (s-vipin) [reply suites/tentacle/rados/tier-2_rados_test_osd_deployment_workflows.yaml:180]: #     comments: Intermittent bug 2357894

File: suites/tentacle/rados/tier-2_rados_test_osd_ok_to_upgrade.yaml [hunk 4]
```diff
@@ -365,3 +365,15 @@ tests:
         erasure_coded_pool: true
         ec_k: 2
         ec_m: 2
+
+  - test:
+      name: ok-to-upgrade-negative-scenarios
+      desc: >-
```
Comment (s-vipin) [inline suites/tentacle/rados/tier-2_rados_test_osd_ok_to_upgrade.yaml]: Fix description
Credits:- @harshkumarRH

File: tests/rados/test_osd_ok_to_upgrade_holistic_tests.py [hunk 5]
```diff
@@ -457,62 +456,89 @@ def run(ceph_cluster, **kw):
...
+                for line in ok_to_stop_lines:
+                    log.info(line)
+                log.info(f"---end of ok-to-stop log lines on {mgr_hostname}---")
 
-            log.info(
-                "ok-to-stop command will be used for all workflows which does not include"
-                " –-bucket_type=<chassis/rack> --bucket_name=<rackOne>"
-            )
+                if ok_to_stop_lines:
+                    ok_to_stop_host_count += 1
```
Comment (s-vipin) [inline tests/rados/test_osd_ok_to_upgrade_holistic_tests.py:500]: Compare mgr that has log lines with active mgr before upgrade. Add log warning 
- @harshkumarRH

Comment (s-vipin) [reply tests/rados/test_osd_ok_to_upgrade_holistic_tests.py:500]: If there is mismatch , print ceph mgr stat

File: suites/tentacle/rados/tier-2_rados_test-bugfixes.yaml (hunk 2 above)
Comment (s-vipin) [reply suites/tentacle/rados/tier-2_rados_test-bugfixes.yaml]: Updated as per comment

**Comment by s-vipin** [comment]
Pass logs:- http://magna002.ceph.redhat.com/cephci-jenkins/vipin-runs/vips_automation_20260729-091839/ 
Please note negative scenarios are failing due to change in exit code. 
earlier it was returning exit code 0 and now returns exit code 22. Needs update in handling. I will pick this as part of next task


```
rhel10@sha256:0f293873acd4ce40397ef9cb9507320e66468ac2bd327f46061c935eaa849964 --daemon-types osd --hosts invalidhost returned Error EINVAL: Upgrade aborted - hosts parameter "invalidhost" provided did not match any hosts
 and code 22 on ceph-vipin-tfa-92-414pvj-node10 [10.0.66.62]
Traceback (most recent call last):
  File "/Users/vipinms/tfa-21-july/tests/rados/test_osd_ok_to_upgrade_holistic_tests.py", line 268, in run
    rados_obj.client.exec_command(cmd=cmd)
  File "/Users/vipinms/tfa-21-july/ceph/ceph.py", line 2135, in exec_command
    raise CommandFailed(
ceph.ceph.CommandFailed: ceph orch upgrade start --image cp.stg.icr.io/cp/ibm-ceph/ceph-9-rhel10@sha256:0f293873acd4ce40397ef9cb9507320e66468ac2bd327f46061c935eaa849964 --daemon-types osd --hosts invalidhost returned Error EINVAL: Upgrade aborted - hosts parameter "invalidhost" provided did not match any hosts
 and code 22 on ceph-vipin-tfa-92-414pvj-node10 [10.0.66.62]
```

**Comment by s-vipin** [comment]
Pass logs for positive scenarios :- http://magna002.ceph.redhat.com/cephci-jenkins/vipin-runs/vips_automation_20260729-091839/
Pass logs for negative scenarios http://magna002.ceph.redhat.com/cephci-jenkins/vipin-runs/vips_automation_20260730-080026/

**Review by harshkumarRH — APPROVED** [review]
Looks good

**Review by pdhiran — APPROVED** [review]
LGTM
